domdiv/__init__.py

```diff
@@ -383,73 +487,263 @@ def baseIndex(self, name):
+                        # this is a designated card to represent the group, so update important data
+                        group_cards[card.group_tag].cost = card.cost
+                        group_cards[card.group_tag].potcost = card.potcost
+                        group_cards[card.group_tag].debtcost = card.debtcost
+                        group_cards[card.group_tag].types = card.types
+                        group_cards[card.group_tag].image = card.image
+
+                    group_cards[card.group_tag].addCardCount(card.count)    # increase the count
+                    # group_cards[card.group_tag].set_lowest_cost(card)  # set holder to lowest cost of the two cards
+
+        cards = keep_cards
+
+        # Now fix up card costs
+        for card in cards:
+            if card.card_tag in group_cards:
+                if group_cards[card.group_tag].isEvent():
+                        group_cards[card.group_tag].cost = "*"
+                        group_cards[card.group_tag].debtcost = 0
+                        group_cards[card.group_tag].potcost = 0
+                if group_cards[card.group_tag].isLandmark():
+                        group_cards[card.group_tag].cost = ""
+                        group_cards[card.group_tag].debtcost = 0
+                        group_cards[card.group_tag].potcost = 0
+
+    # Get the final type names in the requested language
+    Card.type_names = add_type_text(options, Card.type_names, LANGUAGE_DEFAULT)
+    if options.language != LANGUAGE_DEFAULT:
+        Card.type_names = add_type_text(options, Card.type_names, options.language)
+    for card in cards:
+        card.types_name = ' - '.join([Card.type_names[t] for t in card.types]).upper()
+
+    # Get the card bonus keywords in the requested language
+    bonus = add_bonus_regex(options, LANGUAGE_DEFAULT)
+    Card.addBonusRegex(bonus)
+    if options.language != LANGUAGE_DEFAULT:
+        bonus = add_bonus_regex(options, options.language)
+        Card.addBonusRegex(bonus)
+
+    # Fix up cardset text.  Waited as long as possible.
+    Card.sets = add_set_text(options, Card.sets, LANGUAGE_DEFAULT)
+    if options.language != LANGUAGE_DEFAULT:
+        Card.sets = add_set_text(options, Card.sets, options.language)
+
+    for card in cards:
+        if card.cardset_tag in Card.sets:
+            if 'set_name' in Card.sets[card.cardset_tag].keys():
+                card.cardset = Card.sets[card.cardset_tag]['set_name']
+
+    # If expansion names given, then remove any cards not in those expansions
+    # Expansion names can be the names from the language or the cardset_tag
     if options.expansions:
+        options.expansions = list(set(options.expansions))
         options.expansions = [o.lower() for o in options.expansions]
-        reverseMapping = {v: k for k, v in Card.language_mapping.iteritems()}
+        reverseMapping = {set_tag.lower(): Card.sets[set_tag]['set_name'].lower for set_tag in Card.sets}
```

Comment:
I know.  I found it this morning too, but did not have time to fix before work.  I'll do a quick update to fix this.  But after looking at this code again, it still has problems.  For example, it will show an error (but work) if you enter 'dominion' since none of the sets are actually called just 'dominion' any more.  

So I hope to have a real update for this part of the code late Friday or on Saturday morning.